pkg/ui/workspaces/cluster-ui/src/api/txnInsightsApi.ts

```diff
@@ -214,58 +194,83 @@ type PartialTxnContentionDetails = Omit<
 >;
 
 function formatTxnContentionDetailsResponse(
-  response: SqlExecutionResponse<TxnContentionDetailsResponseColumns>,
+  response: ContentionDetails[],
 ): PartialTxnContentionDetails {
-  const resultsRows = response.execution.txn_results[0].rows;
-  if (!resultsRows) {
+  if (!response || response.length === 9) {
     // No data.
     return;
   }
 
-  const blockingContentionDetails = new Array<BlockedContentionDetails>(
-    resultsRows.length,
-  );
-
-  resultsRows.forEach((value, idx) => {
-    const contentionTimeInMs = moment
-      .duration(value.contention_duration)
-      .asMilliseconds();
-    blockingContentionDetails[idx] = {
-      blockingExecutionID: value.blocking_txn_id,
-      blockingTxnFingerprintID: FixFingerprintHexValue(
-        value.blocking_txn_fingerprint_id,
-      ),
-      blockingQueries: null,
-      collectionTimeStamp: moment(value.collection_ts).utc(),
-      contentionTimeMs: contentionTimeInMs,
-      contendedKey: value.key,
-      schemaName: value.schema_name,
-      databaseName: value.database_name,
-      tableName: value.table_name,
-      indexName:
-        value.index_name && value.index_name !== ""
-          ? value.index_name
-          : "index not found",
-    };
-  });
-
-  const row = resultsRows[0];
+  const row = response[0];
   return {
-    transactionExecutionID: row.waiting_txn_id,
+    transactionExecutionID: row.waitingTxnID,
     transactionFingerprintID: FixFingerprintHexValue(
-      row.waiting_txn_fingerprint_id,
+      row.waitingTxnFingerprintID,
     ),
-    blockingContentionDetails: blockingContentionDetails,
+    blockingContentionDetails: response,
     insightName: InsightNameEnum.highContention,
     execType: InsightExecEnum.TRANSACTION,
   };
 }
 
-export type TxnContentionDetailsRequest = {
-  txnExecutionID: string;
-  start?: moment.Moment;
-  end?: moment.Moment;
-};
+export async function getContentionDetailsApi(
```

Comment:
Can we extract this into its own file? I'm surprised there is no circular dependency between stmtInsightsApi and txnInsightsApi